app/models/challenge.rb

```diff
@@ -34,8 +34,16 @@ def challenge_open?
     state.eql? 'open'
   end
 
+  def before_close?
+    unless challenge_end.nil?
+      time = Time.now.utc
+      return time < challenge_end
+    end
```

Comment:
This should be able to be replaced with `return Time.now.utc < challenge_end unless challenge_end.nil?`